File: JoshHarmon.Github/Models/Commit.cs
using System;
namespace JoshHarmon.Github.Models
{
    public class Commit
    {
        public Commit(string committerName, string committerEmail, DateTime dateTime,
            string message, string sha, string url)
        {
            CommitterName = committerName ?? throw new ArgumentNullException(nameof(committerName));
            CommitterEmail = committerEmail ?? throw new ArgumentNullException(nameof(committerEmail));
            DateTime = dateTime;
            Message = message ?? throw new ArgumentNullException(nameof(message));
            Sha = sha ?? throw new ArgumentNullException(nameof(sha));
            Url = url ?? throw new ArgumentNullException(nameof(url));
        }

        public string CommitterName { get; }

        public string CommitterEmail { get; }

        public DateTime DateTime { get; }

        public string Message { get; }

        public string Sha { get; }

        public string Url { get; }
    }
}

File: JoshHarmon.Site/ClientApp/src/components/Projects/Commit.js
import React, { Component } from "react";

export class Commit extends Component {
    static displayName = Commit.name;

    constructor(props) {
        super(props);
        this.state = {
            name: props.model.committerName,
            email: props.model.committerEmail,
            dateTime: props.model.dateTime,
            message: props.model.message,
            sha: props.model.sha,
            url: props.model.url
        };
    }

    render() {
        return (
            <div className='commit m2 p5'>
                <a href={this.state.url}>
                    <p className="m0">{this.state.message}</p>
                    <div className='semi-thin break-word'>
                        <span style={{ minWidth: "fit-content" }}>
                            <b>{this.state.name}</b> <i>({this.state.email})</i>
                        </span>
                    </div>
                </a>
            </div>
        );
    }
}
